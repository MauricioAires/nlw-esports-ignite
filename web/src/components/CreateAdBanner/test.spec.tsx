import { render, screen } from '@testing-library/react';
import useEvent from '@testing-library/user-event';

import * as Dialog from '@radix-ui/react-dialog';

import { CreateAdBanner } from '.';

const sut = () =>
  render(
    <Dialog.Root>
      <CreateAdBanner />

      <Dialog.Portal>
        <Dialog.Overlay />
        <Dialog.Content>
          <Dialog.Title>Modal title</Dialog.Title>

          <p>Modal Content</p>
        </Dialog.Content>
      </Dialog.Portal>
    </Dialog.Root>
  );

describe('<CreateAdBanner />', () => {
  it('should render correclty', () => {
    const { container } = sut();
    expect(
      screen.getByRole('heading', {
        name: /Não encontrou seu duo\?/i,
      })
    ).toBeInTheDocument();

    expect(
      screen.getByText(/Publique um anúncio para encontrar novos players!/)
    ).toBeInTheDocument();

    expect(
      screen.getByRole('button', {
        name: /Publica anúncio/i,
      })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should open modal when click button', async () => {
    sut();

    expect(
      screen.queryByRole('heading', {
        name: /Modal title/i,
      })
    ).not.toBeInTheDocument();

    await useEvent.click(
      screen.getByRole('button', {
        name: /Publica anúncio/i,
      })
    );

    expect(
      screen.getByRole('heading', {
        name: /Modal title/i,
      })
    ).toBeInTheDocument();
    expect(screen.getByText(/Modal Content/i)).toBeInTheDocument();
  });

  it('should close modal when press [esc]', async () => {
    sut();

    // abrir modal
    await useEvent.click(
      screen.getByRole('button', {
        name: /Publica anúncio/i,
      })
    );

    expect(
      screen.getByRole('heading', {
        name: /Modal title/i,
      })
    ).toBeInTheDocument();

    await useEvent.keyboard('{ESCAPE}');

    expect(
      screen.queryByRole('heading', {
        name: /Modal title/i,
      })
    ).not.toBeInTheDocument();
  });
});
