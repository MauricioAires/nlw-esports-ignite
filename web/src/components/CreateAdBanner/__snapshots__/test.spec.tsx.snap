// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateAdBanner /> should render correclty 1`] = `
<div
  class="pt-1 bg-nlw-gradient mt-8 self-stretch rounded-lg overflow-hidden"
>
  <div
    class="bg-[#2A2634] px-8 py-6 self-stretch flex justify-between items-center"
  >
    <div>
      <h2
        class="text-2xl text-white font-black"
      >
        Não encontrou seu duo?
      </h2>
      <p
        class="text-zinc-400"
      >
        Publique um anúncio para encontrar novos players!
      </p>
    </div>
    <button
      aria-controls="radix-:r0:"
      aria-expanded="false"
      aria-haspopup="dialog"
      class="py-3 px-4 bg-violet-600 hover:bg-violet-700 text-white rounded flex items-center gap-3"
      data-state="closed"
      type="button"
    >
      <svg
        fill="currentColor"
        height="24"
        viewBox="0 0 256 256"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="84"
          x2="148"
          y1="116"
          y2="116"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="116"
          x2="116"
          y1="84"
          y2="148"
        />
        <circle
          cx="116"
          cy="116"
          fill="none"
          r="84"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="175.4"
          x2="224"
          y1="175.4"
          y2="224"
        />
      </svg>
       Publica anúncio
    </button>
  </div>
</div>
`;
